platform/posix/transport/src/openssl_posix.c

```diff
@@ -729,55 +721,96 @@ int32_t Openssl_Recv( NetworkContext_t * pNetworkContext,
 {
     OpensslParams_t * pOpensslParams = NULL;
     int32_t bytesReceived = 0;
-    int32_t sslError = 0;
 
     if( ( pNetworkContext == NULL ) || ( pNetworkContext->pParams == NULL ) )
     {
         LogError( ( "Parameter check failed: pNetworkContext is NULL." ) );
     }
-    else if( pNetworkContext->pParams->pSsl != NULL )
+    else if( pNetworkContext->pParams->pSsl == NULL )
+    {
+        LogError( ( "Failed to receive data over network: "
+                    "SSL object in network context is NULL." ) );
+    }
+    else
     {
+        int32_t pollStatus = 1, readStatus = 1, sslError = 0;
+        uint8_t shouldRead = 0U;
+        struct pollfd pollFds;
         pOpensslParams = pNetworkContext->pParams;
-        /* SSL read of data. */
-        bytesReceived = ( int32_t ) SSL_read( pOpensslParams->pSsl,
-                                              pBuffer,
-                                              ( int32_t ) bytesToRecv );
+
+        /* Initialize the file descriptor. */
+        pollFds.events = POLLIN | POLLPRI | POLLHUP | POLLERR;
```

Comment:
According to the [Linux man page](https://man7.org/linux/man-pages/man2/poll.2.html) for `poll.h`:
"_The bits returned in revents can include any of those specified in events, or one of the values **POLLERR**, **POLLHUP**, or **POLLNVAL**.  (These three bits are meaningless in the events field, and will be set in the revents field whenever the corresponding condition is true.)_"

Thus. explicitly setting `POLLHUP | POLLERR` bits in the `events` member is not required..